becquerel/core/fitting.py

```diff
@@ -836,6 +837,61 @@ def _likelihood_residual(self, params, data, weights, **kwargs):
     def eval(self, x, params=None, **kwargs):
         return self.model.eval(x=x, params=params, **kwargs)
 
+    def calc_area_and_unc(self, x=None, component=None):
+        """Calculate the area (and uncertainty) under the fit (or component thereof).
+
+        Parameters
+        ----------
+        x : array-like, optional
+            x values to use for function evaluation, by default `self.x`
+        component : Model, optional
+            Model component for which to calculate the area, by default None, in which
+            case the entire model is used.
+
+        Returns
+        -------
+        uncertainties.ufloat
+            Area under the fit and its 1-sigma uncertainty.
+
+        References
+        ----------
+        .. Tellinghuisen, J. (2001). Statistical error propagation. The Journal of
+           Physical Chemistry A, 105(15), 3917-3921.
+
+        """
+        import numdifftools as nd
+
+        def _calc_area(param_vec, **kwargs):
+            """Internal function to compute the area given the fit values."""
+            param_dict = {name: val for (name, val) in zip(kwargs["names"], param_vec)}
+            return kwargs["model"].eval(x=kwargs["xvals"], **param_dict).sum()
+
+        # Handle input defaults
+        xvals = self.x if x is None else x
+        model = self.model if component is None else component
+
+        # Reformat best_values info so the gradient calculation can handle _calc_area
+        names = list(self.best_values.keys())
+        values = list(self.best_values.values())
+
+        # Compute the area under the curve
+        area = _calc_area(values, xvals=xvals, model=model, names=names)
+
+        # Compute the gradient with respect to the best fit parameters
+        grad = nd.Gradient(_calc_area)
+        g = np.atleast_2d(grad(values, xvals=xvals, model=model, names=names)).T
```

Comment:
Something goes awry in these lines with `lmfit`. The `g` values differ between `lmfit` and `iminuit`, as does the order of the `names`/`values`. That probably throws a wrench into the `g * cov * g` multiplication, leading to the wrong uncertainty.

Reply:
The issue is with `lmfit`:
```py
>>> list(fitter.result.best_values.keys())
['line_m',
 'line_b',
 'gauss1_amp',
 'gauss1_mu',
 'gauss1_sigma',
 'gauss0_amp',
 'gauss0_mu',
 'gauss0_sigma']

>>> list(fitter.params)
['gauss0_amp',
 'gauss0_mu',
 'gauss0_sigma',
 'gauss1_amp',
 'gauss1_mu',
 'gauss1_sigma',
 'line_m',
 'line_b',
 'gauss0_fwhm',
 'gauss1_fwhm']
```

We can fix this by using `fitter.param_names` (which works for both backends) and filtering out the `_fwhm` parameters:
```py
names = [name for name in self.param_names if self.params[name].vary]
values = [self.param_val(name) for name in names]
```